sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs

```diff
@@ -119,7 +119,9 @@ public virtual IEnumerable<Response<KeyBase>> GetKeys(CancellationToken cancella
 
         public virtual IAsyncEnumerable<Response<KeyBase>> GetKeysAsync(CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            Uri firstPageUri = new Uri(_vaultUri, KeysPath + $"?api-version={ApiVersion}");
```

Comment:
The API version argument may be a good candidate to turn into a class-level `const` or `readonly` member;  it seems to be set via constant value in the class and doesn't appear to vary from call-to-call.   That would let you pay the cost of formatting the slug once upfront.

If not, you may want to consider pulling the `KeysPath` into the bling string.   Unless I'm mistaken, the rough translation for the current pattern is something like:

```c#
KeysPath.Concat(String.Format("?api-version{0}", ApiVersion));
```
_(note: someone should verify me here;  there may well be a compiler optimization that I'm not aware of that does a more efficient translation.)_

Reply:
We need to get rid of the string formatting as part for Uri construction, UriBuilder is way for efficient for this.

Reply:
True, but unless I'm overlooking something, if we hoist the constant part up to a class member and reuse, then we're just passing a set of arguments to the standard `Uri` constructor.   Do you still think that there would be a need for `UriBuilder` in that case?

Reply:
I am making a change to use `HttpPipelineUriBuilder` and making it a method so that it is not repeated everywhere

Reply:
@jsquire it works in the case where the path doesn't contain dynamic parts (like a key name) but I would rather have one way generating URIs for all methods.